dev/AppNotifications/AppNotification.cpp

```diff
@@ -0,0 +1,102 @@
+﻿#include "pch.h"
+#include "AppNotification.h"
+#include "Microsoft.Windows.AppNotifications.AppNotification.g.cpp"
+
+using namespace winrt::Windows::Data::Xml::Dom;
+
+namespace winrt::Microsoft::Windows::AppNotifications::implementation
+{
+    AppNotification::AppNotification(hstring const& payload)
+    {
+        XmlDocument xmlDocument{};
+
+        // We call LoadXml to verify the payload is xml
+        xmlDocument.LoadXml(payload);
+        m_payload = payload;
+    }
+
+    hstring AppNotification::Tag()
+    {
+        return m_tag;
+    }
+
+    void AppNotification::Tag(hstring const& tag)
```

Comment:
>AppNotification [](http://example.com/codeflow?start=9&length=15)

This type is marked as "both" meaning "agile" but there's no read/write locks anywhere in this type. Should there be?

Reply:
This needs locks.